instrumentation/okhttp/okhttp-3.0/library/src/main/java/io/opentelemetry/instrumentation/library/okhttp/v3_0/OkHttpInstrumentation.java

```diff
@@ -109,13 +114,21 @@ public static PeerServiceResolver newPeerServiceResolver() {
      * href="https://github.com/open-telemetry/semantic-conventions/blob/main/specification/metrics/semantic_conventions/http-metrics.md#metric-httpclientresponsesize">
      * the response size</a>.
      */
-    public static void setEmitExperimentalHttpClientMetrics(
-            boolean emitExperimentalHttpClientMetrics) {
-        OkHttpInstrumentationConfig.emitExperimentalHttpClientMetrics =
-                emitExperimentalHttpClientMetrics;
+    public void setEmitExperimentalHttpClientMetrics(boolean emitExperimentalHttpClientMetrics) {
+        this.emitExperimentalHttpClientMetrics = emitExperimentalHttpClientMetrics;
     }
 
-    public static boolean emitExperimentalHttpClientMetrics() {
+    public boolean emitExperimentalHttpClientMetrics() {
         return emitExperimentalHttpClientMetrics;
     }
+
+    public OpenTelemetryRum getOpenTelemetryRum() {
+        return openTelemetryRum;
+    }
+
+    @Override
+    public void install(
+            @NotNull Application application, @NotNull OpenTelemetryRum openTelemetryRum) {
+        this.openTelemetryRum = openTelemetryRum;
+    }
```

Comment:
Hmmmm, this feels weird to me. Installing does nothing but hold onto the instance, so that we can add a get() method to this instrumentation so that the lazy singletons can get at the instance. Shouldn't the `install()` method actually help do the creational work? 

The lifecycle is further complicated / obfuscated by the lazy caching stuff in the singletons class.

Here's an idea: What if the `install()` method just passed the needed otel instance to a static factory method in the singletons class, and that is responsible for setting up the instrumenter and the dependent interceptors?

I made a demonstration PR to this branch to show you more what I'm talking about: https://github.com/LikeTheSalad/opentelemetry-android/pull/1

Note that in addition to the lifecycle being somewhat clearer, it also allows us to remove the otelRum instance, remove the lazy interceptor, and the cached supplier.

Reply:
Got it, thanks! I took a quick look and your approach looks a lot cleaner, although I'm concerned about removing the _lazyness_ of the singletons due to the timing between the instantiation of OkHttpClient objects and the OTel rum initialization. I remember it was a key issue when using GlobalOpenTelemetry, though maybe it's no longer the case. I'd like to discuss it further in the SIG just to make sure.

Reply:
Thanks. I've applied the changes suggested [here](https://github.com/LikeTheSalad/opentelemetry-android/pull/1) taking into account what was discussed in the SIG meeting as well 👍